qemu/tests/watchdog.py

```diff
@@ -168,8 +169,12 @@ def check_watchdog_support():
         """
         qemu_binary = utils_misc.get_qemu_binary(params)
 
-        watchdog_type_check = params.get(
-            "watchdog_type_check", " -watchdog '?'")
+        if "aarch" in platform.machine():
+            machine_type = params.get("machine_type").split(':', 1)[1]
```

Comment:
The parameter `machine_type` should not be None here, so `params['machine_type']` would be better. And `[-1]` should be safer than `[1]` in case there is no `:` there.

Reply:
> The parameter `machine_type` should not be None here, so `params['machine_type']` would be better. And `[-1]` should be safer than `[1]` in case there is no `:` there.
Yes, you are right!   Thanks for your suggest！